File: tests/Generator/resource/swift/complex/StructDefinitionType.swift
// A struct represents a class/structure with a fix set of defined properties.
class StructDefinitionType: DefinitionType {
    var base: Bool?
    var discriminator: String?
    var mapping: Dictionary<String, String>?
    var parent: ReferencePropertyType?
    var properties: Dictionary<String, PropertyType>?

    enum CodingKeys: String, CodingKey {
        case base = "base"
        case discriminator = "discriminator"
        case mapping = "mapping"
        case parent = "parent"
        case properties = "properties"
    }
}

File: tests/Generator/resource/csharp/complex/StructDefinitionType.cs
using System.Text.Json.Serialization;

namespace TypeAPI.Model;

/// <summary>
/// A struct represents a class/structure with a fix set of defined properties.
/// </summary>
public class StructDefinitionType : DefinitionType
{
    [JsonPropertyName("base")]
    public bool? Base { get; set; }

    [JsonPropertyName("discriminator")]
    public string? Discriminator { get; set; }

    [JsonPropertyName("mapping")]
    public System.Collections.Generic.Dictionary<string, string>? Mapping { get; set; }

    [JsonPropertyName("parent")]
    public ReferencePropertyType? Parent { get; set; }

    [JsonPropertyName("properties")]
    public System.Collections.Generic.Dictionary<string, PropertyType>? Properties { get; set; }

}

